File: collectors/ruby/inventory.rb
#!/usr/bin/env ruby
# collectors/ruby/inventory.rb
# Version: 0.6.0
# Usage: ruby inventory.rb <workdir>
# Prints inventory to stdout.

workdir = ARGV[0] || "."
ts = `date -u '+%Y-%m-%dT%H:%M:%SZ' 2>/dev/null`.strip
uname = `uname -a 2>/dev/null`.strip

puts "ruby_inventory_version=0.6.0"
puts "workdir=#{workdir}"
puts "ts_utc=#{ts}" unless ts.empty?
puts "uname=#{uname}" unless uname.empty?
puts "ruby=#{RUBY_VERSION}"

File: collectors/perl/inventory.pl
#!/usr/bin/env perl
# collectors/perl/inventory.pl
# Version: 0.6.0
# Usage: perl inventory.pl <workdir>
# Prints inventory to stdout.

use warnings;

my $workdir = $ARGV[0] // ".";
chomp(my $ts = `date -u '+%Y-%m-%dT%H:%M:%SZ' 2>/dev/null`);
chomp(my $uname = `uname -a 2>/dev/null`);

print "perl_inventory_version=0.6.0\n";
print "workdir=$workdir\n";
print "ts_utc=$ts\n" if $ts;
print "uname=$uname\n" if $uname;
print "perl=$^V\n";
